torch/random.py

```diff
@@ -60,9 +62,37 @@ def seed() -> int:
     if not torch.mps._is_in_bad_fork():
         torch.mps.manual_seed(seed)
 
+    _seed_custom_device(seed)
+
     return seed
 
 
+def _seed_custom_device(seed) -> None:
+    r"""Sets the seed to generate random numbers for custom device.
+
+    Args:
+        seed (int): The desired seed.
+
+    Note: Only used for the custom device to set the seed, and you must
+    have to register a custom backend module with `torch._register_device_module("foo", BackendModule)`.
+    BackendModule needs to have the following API's:
+    (1) _is_in_bad_fork() -> bool
+        Return `True` if now it is in bad_fork, else it will return `False`
+
+    (2) manual_seed_all(seed: int) -> None
+        Sets the seed for generating random numbers for your devices.
```

Comment:
It looks like this behavior:

(1) relies on the backend module having some extra API's as mentioned
(2) will silently not do anything, if the backend module doesn't register those API's

One option would be to warn in this case, although I'm not sure if that warning will be too noisy.

For now, can you add a mention of these API's in `torch/utils/backend_registration.py`, where the other centralized info on PrivateUse1 backend registration is? (You can add a note that links to the docs here to avoid duplication)

Reply:
ok I have add a mention, please review again. And maybe I will add give more PR to  enhance scalability for custom device, such as DDP, Libtorch, JIT, Serialization and some on. And maybe I will add more notes, it is ok to put it  in `torch/utils/backend_registration.py`? @bdhirsh

Reply:
For now, I think that's ok. Some time in the near-to-mid future, I think we should update this set of docs with an overview of the PrivateUse1 backend integration! https://pytorch.org/tutorials/advanced/extend_dispatcher.html

Reply:
@bdhirsh  Hi, maybe you forget to merge this pr, please review again.